src/components/FormStep.js

```diff
@@ -121,6 +122,9 @@ const FormStep = ({
   const controller = useRef(new AbortController());
   const configurationRef = useRef(configuration);
 
+  const [shouldFormStepSaveModalBeOpen, setShouldFormStepSaveModalBeOpen] = useState(false);
+  const closeFormStepSaveModal = () => setShouldFormStepSaveModalBeOpen(false);
```

Comment:
we already have an immer reducer for this component, use that instead of adding extra `useState` calls